server/src/main/java/io/crate/planner/operators/InsertFromValues.java

```diff
@@ -634,10 +646,17 @@ private static ShardLocation getShardLocation(String indexName,
         AtomicReference<Throwable> lastFailure = new AtomicReference<>(null);
 
         for (ShardUpsertRequest request : shardUpsertRequests) {
+            String nodeId = state.routingTable()
+                .shardRoutingTable(request.shardId())
+                .primaryShard()
+                .currentNodeId();
+            final ConcurrencyLimit nodeLimit = nodeLimits.get(nodeId);
```

Comment:
Do we even want to use the node limits here for the insert-from-values cases? I'm inclined to leave it as it was.

Reply:
Not sure, but keep it to avoid overloads with concurrent writes running by updates or copy from?